src/components/Preferences/LocalIdentities.js

```diff
@@ -19,132 +20,250 @@ import {
+          new Map(
+            identities.map(({ address }) => [
+              address,
+              !(
+                Array.from(addressesSelected.values()).every(v => v) ||
+                Array.from(addressesSelected.values()).some(v => v)
+              ),
+            ])
+          )
+        ),
+      [addressesSelected, identities]
+    )
+    const handleToggleAddress = React.useCallback(
+      address => () =>
+        setAddressesSelected(
+          new Map([
+            ...addressesSelected,
+            [address, !addressesSelected.get(address)],
+          ])
+        ),
+      [addressesSelected]
+    )
 
-  if (!identities.length) {
-    return <EmptyLocalIdentities onImport={onImport} />
+    React.useEffect(() => {
+      setAddressesSelected(initialAddressesSelected)
+    }, [initialAddressesSelected])
+
+    return (
+      <LocalIdentities
+        {...props}
+        identities={identities}
+        onToggleAll={handleToggleAll}
+        onToggleAddress={handleToggleAddress}
+        addressesSelected={addressesSelected}
+      />
+    )
   }
+)
+
+SelectableLocalIdentities.propTypes = {
+  dao: PropTypes.string.isRequired,
+  localIdentities: PropTypes.object,
+  onClearAll: PropTypes.func.isRequired,
+  onImport: PropTypes.func.isRequired,
+  onModify: PropTypes.func.isRequired,
+  onModifyEvent: PropTypes.func,
+}
+
+SelectableLocalIdentities.defaultProps = {
+  localIdentities: {},
+}
 
-  const { identityEvents$ } = React.useContext(IdentityContext)
-  const { showLocalIdentityModal } = React.useContext(LocalIdentityModalContext)
-  const updateLabel = address => async () => {
-    try {
-      await showLocalIdentityModal(address)
-      // preferences get all
-      onModifyEvent()
-      // for iframe apps
-      identityEvents$.next({ type: identityEventTypes.MODIFY, address })
-    } catch (e) {
-      /* nothing was updated */
+const LocalIdentities = React.memo(
+  ({
+    addressesSelected,
+    dao,
+    identities,
+    onClearAll,
+    onImport,
+    onModify,
+    onModifyEvent,
+    onToggleAddress,
+    onToggleAll,
+  }) => {
+    const { identityEvents$ } = React.useContext(IdentityContext)
+    const { showLocalIdentityModal } = React.useContext(
+      LocalIdentityModalContext
+    )
+    const updateLabel = React.useCallback(address => async () => {
+      try {
+        await showLocalIdentityModal(address)
+        // preferences get all
+        onModifyEvent()
+        // for iframe apps
+        identityEvents$.next({ type: identityEventTypes.MODIFY, address })
+      } catch (e) {
+        /* nothing was updated */
+      }
+    })
+
+    const [allSelected, someSelected] = React.useMemo(
+      () => [
+        Array.from(addressesSelected.values()).every(v => v),
+        Array.from(addressesSelected.values()).some(v => v),
+      ],
+      [addressesSelected]
+    )
+    // standard: https://en.wikipedia.org/wiki/ISO_8601
+    const today = format(Date.now(), 'yyyy-MM-dd')
```

Comment:
We should ensure that `today` stays up to date here, so that if someone leaves this screen opened between two days and saves, the file name reflects it.

A solution could be to add a `setTimeout()` corresponding to the time remaining before the end of the day (so that we only re-render once), but `useNow()` with a high `updateEvery` (every minute?) is probably enough.

Reply:
Or make it a dynamic download using `downloadjs` (like in Finance) and the name is given on click. Thoughts?

Reply:
Yes totally! Is there another one more up to date than `downloadjs`? It’s the kind of library that should follow closely what browsers do, and this one [doesn’t seem to be updated often](https://github.com/rndme/download).

Reply:
Done. Used [`file-saver`](https://github.com/eligrey/FileSaver.js) in 8643741498820e6490e4477eeb5d2aeb2e8f096e